src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,161 +1,492 @@
+  '4-gram': [],
+  '5-gram': [],
+  '6-gram': [],
+  '7-gram': [],
+  '8-gram': [],
+  '9-gram': [],
+  '10-gram': []
 })
 
-// Text cleaning function
-const cleanText = (text) => {
-  return text
-    .toLowerCase()
-    .replace(/[^\w\s]/g, ' ') // Remove special characters except spaces
-    .replace(/\s+/g, ' ') // Replace multiple spaces with single space
-    .trim()
-}
-
-// Generate n-gram keywords
-const generateNGrams = (words, n) => {
-  const ngrams = []
-  for (let i = 0; i <= words.length - n; i++) {
-    const ngram = words.slice(i, i + n).join(' ')
-    ngrams.push(ngram)
-  }
-  return ngrams
+// Generates keywords from input text
+const generateKeywords = () => {
+  const keywords = generateKeywordsUtil(
+    inputText.value,
+    selectedNGrams.value
+  )
+  
+  generatedKeywords.value = keywords
 }
 
-// Generate keywords from input text
-const generateKeywords = () => {
-  if (!inputText.value.trim()) {
-    generatedKeywords.value = {
-      '1-gram': [],
-      '2-gram': [],
-      '3-gram': []
-    }
-    return
-  }
 
-  const cleanedText = cleanText(inputText.value)
-  const words = cleanedText.split(' ').filter(word => word.length > 0)
+watch(generatedKeywords, (newKeywords) => {
+  selectedNGrams.value.forEach(n => {
+    const key = `${n}-gram`
+    keywordTags.value[key] = newKeywords[key] || []
+  })
+}, { deep: true })
 
-  // Generate n-grams and remove duplicates
-  const oneGrams = [...new Set(generateNGrams(words, 1))]
-  const twoGrams = [...new Set(generateNGrams(words, 2))]
-  const threeGrams = [...new Set(generateNGrams(words, 3))]
 
-  generatedKeywords.value = {
-    '1-gram': oneGrams,
-    '2-gram': twoGrams,
-    '3-gram': threeGrams
+const toggleNGram = (n) => {
+  const index = selectedNGrams.value.indexOf(n)
+  if (index > -1) {
+    selectedNGrams.value.splice(index, 1)
+  } else {
+    selectedNGrams.value.push(n)
+    selectedNGrams.value.sort((a, b) => a - b)
   }
+  generateKeywords()
 }
 
-// Computed properties for statistics
+
 const totalKeywords = computed(() => {
-  return generatedKeywords.value['1-gram'].length + 
-         generatedKeywords.value['2-gram'].length + 
-         generatedKeywords.value['3-gram'].length
+  return Object.values(generatedKeywords.value).reduce((total, keywords) => {
+    return total + keywords.length
+  }, 0)
 })
 
 const hasKeywords = computed(() => {
   return totalKeywords.value > 0
 })
+
+
+const nGramOptions = getNGramOptions(10)
 </script>
 
 <template>
-    <div class="max-w-6xl mx-auto p-5 font-sans">
-        <div class="text-center mb-10">
-            <h1 class="text-4xl font-semibold text-gray-800 mb-2">Keyword Generator</h1>
-            <p class="text-lg text-gray-600">Generate n-gram keywords from application descriptions</p>
+    <div class="container">
+        <div class="header">
+            <h1 class="title">Keyword Generator</h1>
+            <p class="subtitle">Generate n-gram keywords from application descriptions</p>
```

Comment:
```suggestion
    <div class="ma-container">
        <div class="ma-header">
            <h1 class="title">Keyword Generator</h1>
            <p class="subtitle">Generate n-gram keywords from application descriptions</p>
```
Please use the ma tag AMAP for styling